Assets/__Scripts/MapEditor/Grid/Collections/EventsContainer.cs

```diff
@@ -18,13 +18,24 @@ public bool RingPropagationEditing
         set
         {
             ringPropagationEditing = value;
-            ringPropagationLabels.SetActive(value);
-            eventGridLabels.SetActive(!value);
+            PlatformDescriptor pD = FindObjectOfType<PlatformDescriptor>();
+            eventGridLabels.transform.parent.GetComponent<CreateEventTypeLabels>().UpdateLabels(value, value ? (pD.BigRingManager?.rings.Length ?? 15)+1 : 16);
+            eventPlacement.SetGridSize(value ? (pD.BigRingManager?.rings.Length ?? 15) + 1 : 6 + pD.LightingManagers.Count(s => s != null));
+
+            //ringPropagationLabels.SetActive(value);
+            //eventGridLabels.SetActive(!value);
             UpdateRingPropagationMode();
         }
     }
     private bool ringPropagationEditing = false;
 
+    public void StartUp()
```

Comment:
a `Start` function exists that is called on Every MonoBehaviour object. Please use that instead.

Reply:
During Start, the PlatformDescriptor is not available (null), I now added a WaitForSeconds(0.1f) like it was done in other scripts, it seems to work. I only hope that this will work every time

Reply:
If your intention is to get the PlatformDescriptor that has spawned, then look into the `LoadInitialMap.PlatformLoadedEvent` event. You'll need to subscribe to it in the Start function and unsubscribe from the OnDestroy function.

Reply:
I changed as you suggested (With `PlatformLoadedEvent`) and resolved the merge conflict, ready to be pulled :)